### PR TITLE
Add max depth of binary tree solution

### PR DESCRIPTION
**Pull request by vlemish**
Runtime: 111 ms, faster than 69.93% of C# online submissions for Maximum Depth of Binary Tree.
Memory Usage: 37.9 MB, less than 58.27% of C# online submissions for Maximum Depth of Binary Tree.

### CODE REVIEW

File: vlemish/MaximumDepthOfBinaryTree.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            this.left = left;
+            this.right = right;
+        }
+
+        public class MaximumDepthOfBinaryTree
+        {
+            public int MaxDepth(TreeNode root) =>
+                MaxDepth(root, 0);
+
+            private int MaxDepth(TreeNode node, int depth)
```
Comment (PShchegolevatykh) [inline vlemish/MaximumDepthOfBinaryTree.cs:28]: Interesting take to keep depth as a second parameter

File: vlemish/MaximumDepthOfBinaryTree.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace _3Sum
```
Comment (PShchegolevatykh) [inline vlemish/MaximumDepthOfBinaryTree.cs:7]: Wrong namespace 😄

**Review by PShchegolevatykh — APPROVED** [review]

File: vlemish/MaximumDepthOfBinaryTree.cs (hunk 2 above)
Comment (vlemish) [reply vlemish/MaximumDepthOfBinaryTree.cs:7]: I was too lazy to create a new project 🙂

**Comment by rokovalenko** [comment]
👍

**Pull request merged**